app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java

```diff
@@ -1,44 +1,104 @@
 package com.github.onedirection.navigation.fragment.map;
 
+import android.graphics.drawable.Drawable;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
 import androidx.fragment.app.Fragment;
 
 import com.github.onedirection.R;
 import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.geometry.LatLng;
 import com.mapbox.mapboxsdk.maps.MapView;
 import com.mapbox.mapboxsdk.maps.MapboxMap;
 import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.plugins.annotation.Symbol;
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolManager;
+import com.mapbox.mapboxsdk.utils.BitmapUtils;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class MapFragment extends Fragment {
 
     private MapView mapView;
+    private MapboxMap mapboxMap;
+    private MarkerSymbolManager markerSymbolManager;
+    private final String SYMBOL_ID = "MARKER_MAP";
+    private final String BUNDLE_ID = "bundleT";
+    private final String MARKERS_ID = "markers_size";
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater,
                              @Nullable ViewGroup container,
                              @Nullable Bundle savedInstanceState) {
-        Mapbox.getInstance(getContext().getApplicationContext(), getString(R.string.mapbox_access_token));
-
+        Mapbox.getInstance(getContext(), getString(R.string.mapbox_access_token));
         View view = inflater.inflate(R.layout.fragment_map, container, false);
 
-        mapView = (MapView) view.findViewById(R.id.mapView);
+        mapView = view.findViewById(R.id.mapView);
         mapView.onCreate(savedInstanceState);
-        mapView.getMapAsync(mapboxMap -> mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+        mapView.getMapAsync(mapboxMap -> {
+            this.mapboxMap = mapboxMap;
+
+            mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+                initializeMarkerSymbolManager(style);
+            });
 
-        }));
+            mapboxMap.addOnMapClickListener(point -> {
+                markerSymbolManager.removeAllMarker();
+                markerSymbolManager.addMarker(point);
+                return false;
+            });
+        });
         return view;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        System.out.println(savedInstanceState);
```

Comment:
Leftover debug.
If you want to keep it:
```suggestion
        private finale static String LOGCAT_ID = "MapFragment";
        ...
        Log.d(LOGCAT_ID, savedInstanceState.toString());
```

Reply:
Removed